### PR TITLE
Move IndexDocument and its subclasses to their own file

### PR DESCRIPTION
**Pull request by mweiden**
`dss/events/handlers/index.py` was growing quite large, having both classes that handle both storage events and index documents. This pull request moves IndexDocument and it's child classes to their own module.

### CODE REVIEW

File: tests/test_indexer.py [hunk 1]
```diff
@@ -12,10 +14,6 @@
 import uuid
 from http.server import BaseHTTPRequestHandler, HTTPServer
 
-import google.auth
```
Comment (hannes-ucsc) [inline tests/test_indexer.py]: Nit: I'm totally fine with sneaking in such small fixes into an unrelated PR as long as they are in a separate commit AND the PR lands unsquashed.

File: dss/storage/index_document.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+            self.logger.info(f"Successfully notified for subscription {subscription_id}"
+                             f" for bundle {self.bundle_id} with transaction id {transaction_id} "
+                             f"Code: {response.status_code}")
+        else:
+            self.logger.warning(f"Failed notification for subscription {subscription_id}"
+                                f" for bundle {self.bundle_id} with transaction id {transaction_id} "
+                                f"Code: {response.status_code}")
+
+
+class BundleTombstoneDocument(IndexDocument):
```
Comment (hannes-ucsc) [inline dss/storage/index_document.py]: As much as I would like to merge this quickly, but this should not be introduced in this PR.

File: dss/storage/index_document.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+        blobstore, _, bucket_name = Config.get_cloud_specific_handles(replica.name)
+        bundle_uuid, bundle_version = DSS_OBJECT_NAME_REGEX.search(key).groups()
+
+        tombstone_data = json.loads(blobstore.get(bucket_name, key))
+
+        doc = cls(replica, bundle_uuid, bundle_version, logger)
+        doc.update(tombstone_data)
+        return doc
+
+    def list_dead_bundles(self):
```
Comment (hannes-ucsc) [inline dss/storage/index_document.py]: Love this name.

File: dss/storage/index_document.py (hunk 2 above)
Comment (hannes-ucsc) [reply dss/storage/index_document.py]: My mistake.

**Comment by mweiden** [comment]
@hannes-ucsc ty hannes!

**Pull request merged**